src/mlpack/core/math/lin_alg_impl.hpp

```diff
@@ -22,6 +25,268 @@ inline size_t SvecIndex(size_t i, size_t j, size_t n)
   return (j-i) + (n*(n+1) - (n-i)*(n-i+1))/2;
 }
 
+/**
+ * Auxiliary function to raise vector elements to a specific power.  The sign
+ * is ignored in the power operation and then re-added.  Useful for
+ * eigenvalues.
+ */
+inline void VectorPower(arma::vec& vec, const double power)
+{
+  for (size_t i = 0; i < vec.n_elem; ++i)
+  {
+    if (std::abs(vec(i)) > 1e-12)
+      vec(i) = (vec(i) > 0) ? std::pow(vec(i), (double) power) :
+          -std::pow(-vec(i), (double) power);
+    else
+      vec(i) = 0;
+  }
+}
+
+/**
+ * Creates a centered matrix, where centering is done by subtracting
+ * the sum over the columns (a column vector) from each column of the matrix.
+ *
+ * @param x Input matrix
+ * @param xCentered Matrix to write centered output into
+ */
+inline void Center(const arma::mat& x, arma::mat& xCentered)
+{
+  // Get the mean of the elements in each row.
+  arma::vec rowMean = arma::sum(x, 1) / x.n_cols;
+
+  xCentered = x - arma::repmat(rowMean, 1, x.n_cols);
+}
+
+/**
+ * Whitens a matrix using the singular value decomposition of the covariance
+ * matrix. Whitening means the covariance matrix of the result is the identity
+ * matrix.
+ */
+inline void WhitenUsingSVD(const arma::mat& x,
+                                  arma::mat& xWhitened,
+                                  arma::mat& whiteningMatrix)
+{
+  arma::mat covX, u, v, invSMatrix, temp1;
+  arma::vec sVector;
+
+  covX = ColumnCovariance(x);
+
+  svd(u, sVector, v, covX);
+
+  size_t d = sVector.n_elem;
+  invSMatrix.zeros(d, d);
+  invSMatrix.diag() = 1 / sqrt(sVector);
+
+  whiteningMatrix = v * invSMatrix * trans(u);
+
+  xWhitened = whiteningMatrix * x;
+}
+
+/**
+ * Overwrites a dimension-N vector to a random vector on the unit sphere in R^N.
+ */
+inline void RandVector(arma::vec& v)
+{
+  v.zeros();
```

Comment:
Looks like this isn't necessary.

Reply:
Agreed, I think we can remove this line (I know it was not part of the purpose of your PR, but it's a cleanup we could do anyway).